game/game/api/friends.py

```diff
@@ -0,0 +1,156 @@
+        # TODO SysMsg: S1_REQUESTED_TO_BECOME_FRIENDS
+
+
+@l2_request_handler(
+    GAME_REQUEST_FRIEND_INVITE_ANSWER,
+    Template([Parameter(id="answer_type", start=0, length=4, type=ctype.int32)]),
+)
+async def friend_invite_answer(request: GameRequest):
+    friend_character = request.session.character
+    requestor_id = friend_character.active_requestor
+    requstor_character = WORLD.get_character_by_id(requestor_id)
+    requstor_session = requstor_character.session
+
+    answer_type = request.validated_data["answer_type"]
+    if answer_type == 1:
+        friend_character.friends.append(requestor_id)
+        requstor_character.friends.append(friend_character.id)
+
+        friend_character.active_requestor = 0  # to be discussed
+
+        await friend_character.commit_changes(fields=["friends"])
+        await requstor_character.commit_changes(fields=["friends"])
+
+        await friend_character.notify_friends(request.session)
+        # TODO: SysMsg: S1_JOINED_AS_FRIEND
+        await requstor_character.notify_friends(requstor_session)
+        # TODO: SysMsg: S1_ADDED_TO_FRIENDS
+
+        LOG.info(
+            f"{friend_character.name} has accepted friend invite from {requstor_character.name}"
+        )
+
+    else:
+        LOG.info(
+            f"{friend_character.name} failed to accept friend invite from {requstor_character.name}"
+        )
+        # TODO SysMsg: FAILED_TO_INVITE_A_FRIEND
+
+
+@l2_request_handler(
+    GAME_REQUEST_FRIEND_MESSAGE,
+    Template(
+        [
+            Parameter(id="message", start=0, type=str, func=decode_str()),
+            Parameter(id="recipient_name", start="$message.stop", type=str, func=decode_str()),
+        ]
+    ),
+)
+async def send_friend_message(request: GameRequest):
+    name = request.session.character.name
+    message = request.validated_data["message"]
+    recipient_name = request.validated_data["recipient_name"]
+
+    recipient_session = WORLD.get_session_by_character_name(recipient_name)
+    if recipient_session is None:
+        LOG.debug(
+            f"{name} has tried to send friend message {message} to the offline player: {recipient_name}"
+        )
+        # TODO: SysMsg: TARGET_IS_NOT_FOUND_IN_THE_GAME
+        return
+
+    recipient_session.send_packet(
+        FriendMessage(sender_name=name, message=message, recipient_name=recipient_name)
+    )
+    LOG.info(f'{name} has send friend message: "{message}" to {recipient_name}')
+
+
+@l2_request_handler(
+    GAME_REQUEST_FRIEND_DELETE,
+    Template([Parameter(id="friend_name", start=0, type=str, func=decode_str())]),
+)
+async def friend_delete(request: GameRequest):
+    friend_to_delete_name = request.validated_data["friend_name"]
+    if friend_to_delete_name is None:
+        return
+
+    character = request.session.character
+    friend_character = await character.one_by_name(friend_to_delete_name, required=False)
```

Comment:
let's import `Character` class and use its classmethod. 
Using it on an instance of a class is a bit confusing

Reply:
Also in other places

Reply:
Fixed in this specific place. What other places do you mean?